### PR TITLE
Refactor `Generator` and `ManifestGraphLoader`

### PR DESCRIPTION
**Pull request by adellibovi**
### Short description 📝

This PR refactors and simplifies both `Generator` and `ManifestGraphLoader`.
We had some dead code that got removed (the one related with `projectOnly` parameter).
Plus, the branching for loading project and workspace it was converged with mapping of a `Workspace` when wanting to load a project, and instead of duplicating inside `Generator` and `ManifestGraphLoader` it is now only under the latter.

### How to test the changes locally 🧐

Unit tests and acceptances should cover this already.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Review by kwridan — DISMISSED** [review]
Thanks for the much needed clean ups in this area @adellibovi !

This sadly is a breaking change for us  😞 - we leverage the project only option - see the original rationale for this feature in https://github.com/tuist/tuist/issues/452 - we are no longer in the migration phase, but have found this option to be useful when iterating on newly added projects to scope generation to that one project.

If we strongly feel it's not a good fit for Tuist, we can keep it in the `GeneratorDescriptor` and `GraphLoader` APIs _(which we use)_ and remove it from Tuist's public command line interface.

**Comment by adellibovi** [comment]
> This sadly is a breaking change for us 😞 - we leverage the project only option - see the original rationale for this feature in #452 - we are no longer in the migration phase, but have found this option to be useful when iterating on newly added projects to scope generation to that one project.

Hi @kwridan, are you referring to the `--project-only` option? I believe that was already removed (pre-PR we were passing always false) 😕 what I did here was to remove the "dead branch". Or maybe am I missing something else? Thanks 🙏

**Comment by danyf90** [comment]
Oops, looking at the history, it might be that I have removed it by mistake in [this PR](https://github.com/tuist/tuist/pull/3912) 😢 
Good news is that it is part of 3.0 so it has not been released yet, se we can add it back before the release :D

Need to think about how it should work with focus, as hashes of a target depend on its dependencies 🤔   maybe we should make it available only when `--no-cache` is passed?

File: Sources/TuistKit/Generator/Generator.swift [hunk 1]
```diff
@@ -196,30 +74,12 @@ class Generator: Generating {
...
-            workspaceName: workspaceDescriptor.xcworkspacePath.basename
-        )
+    func load(path: AbsolutePath) async throws -> Graph {
+        try await load(path: path).0
+    }
 
-        return (workspaceDescriptor.xcworkspacePath, graph)
+    func load(path: AbsolutePath) async throws -> (Graph, [SideEffectDescriptor]) {
+        try await manifestGraphLoader.load(path: path)
     }
```
Comment (danyf90) [inline Sources/TuistKit/Generator/Generator.swift:83]: Is it used only internally? can we remove it and just call manifestGraphLoader above?

**Review by danyf90 — APPROVED** [review]
GTM, but I would add a comment to explain why we need to keep `generateProject` for now.
Also, in the long term who need it could use the `ProjectDescriptorGenerator` instead of the `Generator, isn't it? 🤔

**Pull request merged**